kairon/chat/utils.py

```
@@ -330,3 +333,17 @@ def add_telemetry_metadata(x_telemetry_uid: Text, x_telemetry_sid: Text, metadat
             metadata["telemetry-sid"] = x_telemetry_sid
         return metadata
 
+    @staticmethod
+    async def handle_media_agentic_flow(bot: str, sender_id: str, name: str, files: List[File], slot_vals: str = None):
+        mids, _ = await UserMedia.upload_media_content_sync(bot, sender_id, files)
+        slots = {}
+        if slot_vals:
+            try:
+                slots = json.loads(slot_vals)
+            except json.JSONDecodeError:
+                raise AppException("Invalid slot values format. Must be a valid JSON string.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)